File: Java/JavaTest/src/main/java/com/cocoa/validation/Person.java
package com.cocoa.validation;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

public class Person extends  BaseClass{

    @NotBlank(message = "name is null!")
    private String name;

    @NotNull(message = "age cannot be null")
    private Integer age;

    @NotNull(message = "address cannot be null")
    private String address;


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
}

File: Cpp/base/class/Person.cpp
#include <iostream>
#include <string>

using namespace std;

class Person{
    public:
        int age;
        string name;

    void show(){
        cout<< "name is " << name << age << endl; 
    }
    void setName(string nName){
        name = nName;
    }

};



int main(){


    Person zhangsan ;
    zhangsan.setName("zhangsan");
    zhangsan.age = 12;
    zhangsan.show();
    return 0;
}


